src/components/Map/algorithms/Chart.tsx

```diff
@@ -0,0 +1,134 @@
+import React from 'react';
+import {
+  Chart as ChartJS,
+  CategoryScale,
+  LinearScale,
+  PointElement,
+  LineElement,
+  Title,
+  Tooltip,
+  Filler,
+  Legend,
+} from 'chart.js';
+import { Line } from 'react-chartjs-2';
+
+ChartJS.register(
+  CategoryScale,
+  LinearScale,
+  PointElement,
+  LineElement,
+  Title,
+  Tooltip,
+  Filler,
+  Legend
+);
+
+export const options = {
+  responsive: true,
+  plugins: {
+    legend: {
+      position: 'top' as const,
+    },
+    title: {
+      display: true,
+      text: 'Diffusion Centrality per node over time',
+    },
+  },
+};
+
+
+function convertToDataFormat() {
+  // data: Map<string, Map<string, Map<string, number>>>
```

Comment:
This is what I expect would come from redux store. But if you isolate:
```js
const diffcen = {...}
```
and pass it to convertToDataFormat, you will get errors saying the type doesn't have certain types of functionalities.

Reply:
what errors specifically?